gen/gen.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"text/template"
+
+	"github.com/fatih/structs"
+	mondoov1 "go.mondoo.com/mondoo-go"
+)
+
+func main() {
+	// Generate resources for Mondoo integrations only
+	err := generateIntegrationResources()
+	if err != nil {
+		log.Fatalln(err)
+	}
+}
+
+type IntegrationResource struct {
+	ResourceClassName     string
+	TerraformResourceName string
+}
+
+// generateIntegrationResources generates Terraform resources for Mondoo's integrations.
+func generateIntegrationResources() error {
+	// Read the template file
+	resourceTemplateFile := filepath.Join("gen", "templates", "integration_resource.go.tmpl")
+	resourceTmpl, err := template.ParseFiles(resourceTemplateFile)
+	if err != nil {
+		return err
+	}
+
+	testTemplateFile := filepath.Join("gen", "templates", "integration_resource_test.go.tmpl")
+	testTmpl, err := template.ParseFiles(testTemplateFile)
+	if err != nil {
+		return err
+	}
+
+	resourceDotTFTemplateFile := filepath.Join("gen", "templates", "resource.tf.tmpl")
+	resourceTFTmpl, err := template.ParseFiles(resourceDotTFTemplateFile)
+	if err != nil {
+		return err
+	}
+
+	// Ensure the output directory exists
+	outputDirPath := filepath.Join("gen", "generated")
+	if err := os.MkdirAll(outputDirPath, 0755); err != nil {
+		return err
+	}
+
+	i := mondoov1.ClientIntegrationConfigurationInput{}
+	m := structs.Map(i)
+
+	for k := range m {
+		// TODO we know the type and the struct associated to the type, we need
+		// to look it (the struct) and use the same `structs.Map(v)` to list all
```

Comment:
Something reads strange to me.
Perhaps:
```suggestion
		// to look at it (the struct) and use the same `structs.Map(v)` to list all
```